### PR TITLE
Parameterize Requeue Time

### PR DESCRIPTION
**Pull request by DharmitD**
## The issue resolved by this Pull Request:
Resolves https://github.com/opendatahub-io/data-science-pipelines-operator/issues/316

## Description of your changes:

Parameterizing the DSPAReconciler requeue time value whilst setting the default to 2m, to allow for more custom requeue times.

## Testing instructions

Set the variable `REQUEUE_TIME` , run DSPO, create a DSPA, and ensure the DSPA deploys correctly.
## Checklist
- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by dsp-developers** [comment]
A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-437`
An OCP cluster where you are logged in as cluster admin is required.

To use this image run the following:

```bash
cd $(mktemp -d)
git clone git@github.com:opendatahub-io/data-science-pipelines-operator.git
cd data-science-pipelines-operator/
git fetch origin pull/437/head
git checkout -b pullrequest 8dbc52885f7472d89d7fdf50f948e89dd9852203
make deploy IMG="quay.io/opendatahub/data-science-pipelines-operator:pr-437"
```

More instructions [here](https://github.com/opendatahub-io/data-science-pipelines-operator#deploy-dsp-instance) on how to deploy and test a Data Science Pipelines Application.

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-437`

**Review by rimolive — COMMENTED** [review]
/lgtm

File: controllers/dspipeline_controller.go [hunk 1]
```diff
@@ -52,6 +52,7 @@ type DSPAReconciler struct {
 	Log                     logr.Logger
 	TemplatesPath           string
 	MaxConcurrentReconciles int
+	RequeueTime             time.Duration
```
Comment (HumairAK) [inline controllers/dspipeline_controller.go:55]: @gmfrasca raised a concern that we may be bloating this struct

Comment (DharmitD) [reply controllers/dspipeline_controller.go]: Done, updated to get the configs from viper.

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-437`

File: controllers/dspipeline_controller.go [hunk 2]
```diff
@@ -20,7 +20,7 @@ import (
 	"context"
 	"fmt"
 	"sigs.k8s.io/controller-runtime/pkg/controller"
-	"time"
+	// 	"time"
```
Comment (hbelmiro) [inline controllers/dspipeline_controller.go:23]: This comment can be removed.

```suggestion
```

Comment (DharmitD) [reply controllers/dspipeline_controller.go]: Good catch! My bad I didn't realize I hadn't taken this out. Deleted the comment.

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-437`

**Review by gmfrasca — APPROVED** [review]

**Pull request merged**